File: OwnHabits.Domain/Models/Goal.cs
using OwnHabits.Domain.Common;
using OwnHabits.Domain.Models.User;

namespace OwnHabits.Domain.Models;

public class Goal: BaseTextEntity
{
    public DateTime CreatedAt { get; init; } = DateTime.Now;
    public DateTime Deadline { get; set; }
    public Status Status { get; set; } = Status.InProgress;
    public Priority Priority { get; set; } = Priority.Normal;
    public Difficulty Difficulty { get; set; } = Difficulty.Easy;
    public bool Repeatability {get; set; } = false;
    public int GainedExperience { get; set; }
    public int Penalty  { get; set; }
    
    public virtual ICollection<GoalsToSkills> UpgradableSkills { get; set; } = [];
    public virtual ICollection<UserToGoals> UserGoals { get; set; } = [];


}

File: Frontend/own-habits-frontend/src/types/Goal.ts
import type { GoalsToSkills } from "./GoalsToSkills";
import type { UserToGoals } from "./UserToGoals";

export type Status = 'InProgress' | 'Completed' | 'Failed' | 'Cancelled';
export type Priority = 'Low' | 'Normal' | 'High' | 'Urgent';
export type Difficulty = 'Easy' | 'Medium' | 'Hard' | 'Extreme';

export interface Goal {
  id: string;
  userId: string;
  title: string;
  description?: string;
  createdAt: string;
  deadline: string;
  status: Status;
  priority: Priority;
  difficulty: Difficulty;
  repeatability: boolean;
  gainedExperience: number;
  penalty: number;
  
  // Навигационные свойства
  upgradableSkills: GoalsToSkills[];
  userGoals: UserToGoals[];
}